python-avd/pyavd/_eos_designs/structured_config/overlay/router_bgp.py

```diff
@@ -24,457 +26,360 @@ class RouterBgpMixin(Protocol):
+                    name=self.inputs.bgp_peer_groups.evpn_overlay_core.name,
+                    domain_remote=True,
+                    activate=True,
                 )
 
             if self.shared_utils.node_config.evpn_gateway.evpn_l3.enabled:
-                address_family_evpn["neighbor_default"] = {
-                    "next_hop_self_received_evpn_routes": {
-                        "enable": True,
-                        "inter_domain": self.shared_utils.node_config.evpn_gateway.evpn_l3.inter_domain,
-                    },
-                }
+                self.structured_config.router_bgp.address_family_evpn.neighbor_default.next_hop_self_received_evpn_routes._update(
+                    enable=True, inter_domain=self.shared_utils.node_config.evpn_gateway.evpn_l3.inter_domain
+                )
 
         if self.shared_utils.overlay_routing_protocol == "ibgp":
             # TODO: - assess this condition - both can't be true at the same time.
             if self.shared_utils.overlay_evpn_mpls is True and self.shared_utils.overlay_evpn_vxlan is not True:
-                overlay_peer_group = {"name": self.inputs.bgp_peer_groups.mpls_overlay_peers.name, "activate": True}
-                address_family_evpn["neighbor_default"] = {"encapsulation": "mpls"}
+                overlay_peer_group._update(name=self.inputs.bgp_peer_groups.mpls_overlay_peers.name, activate=True)
+                self.structured_config.router_bgp.address_family_evpn.neighbor_default.encapsulation = "mpls"
                 if self.shared_utils.overlay_ler is True:
-                    address_family_evpn["neighbor_default"]["next_hop_self_source_interface"] = "Loopback0"
+                    self.structured_config.router_bgp.address_family_evpn.neighbor_default.next_hop_self_source_interface = "Loopback0"
 
                 if self._is_mpls_server is True:
-                    peer_groups.append({"name": self.inputs.bgp_peer_groups.rr_overlay_peers.name, "activate": True})
+                    peer_groups.append_new(name=self.inputs.bgp_peer_groups.rr_overlay_peers.name, activate=True)
 
             # TODO: this is written for matching either evpn_mpls or evpn_vlxan based for iBGP see if we cannot make this better.
             if self.shared_utils.overlay_vtep is True and self.shared_utils.evpn_role != "server" and overlay_peer_group:
-                overlay_peer_group.update(
-                    {
-                        "route_map_in": "RM-EVPN-SOO-IN",
-                        "route_map_out": "RM-EVPN-SOO-OUT",
-                    },
+                overlay_peer_group._update(
+                    route_map_in="RM-EVPN-SOO-IN",
+                    route_map_out="RM-EVPN-SOO-OUT",
                 )
-        if overlay_peer_group:
-            peer_groups.append(overlay_peer_group)
 
-        if peer_groups:
-            address_family_evpn["peer_groups"] = peer_groups
+        peer_groups.append(overlay_peer_group)
```

Comment:
I think we wish to keep the condition. since there is no need to add the class item if we have not filled out anything.

Reply:
Fixed!